src/mrpro/operators/_SliceProjectionOp.py

```diff
@@ -0,0 +1,449 @@
+                grad_x -= matrix_adjoint.imag @ grad_output[0].imag
+        return grad_x, None, None
+
+
+class SliceGaussian:
+    """Gaussian Slice Profile."""
+
+    def __init__(self, fwhm: float | Tensor):
+        self.fwhm = torch.as_tensor(fwhm)
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.exp(-(x**2) / (0.36 * self.fwhm**2))
+
+
+class SliceSmoothedRect:
+    """Rectangular Slice Profile with smooth flanks.
+
+    The smaller n, the smoother it is. For n<1, the FWHM might be wrong
+
+    """
+
+    def __init__(self, fwhm: float | Tensor, n: float | Tensor):
+        self.n = n
+        self.fwhm = fwhm
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        y = x * 2 / self.fwhm
+        return torch.erf(self.n * (1 - y)) + torch.erf(self.n * (1 + y))
+
+
+class SliceInterpolate:
+    """Slice Profile based on Interpolation of Measured Profile."""
+
+    def __init__(self, xs: Tensor, weights: Tensor):
+        self._xs = xs
+        self._weights = weights
+
+    def __call__(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.as_tensor(np.interp(x, self._xs.numpy(), self._weights.numpy(), 0, 0))
+
+
+TensorFunction: TypeAlias = Callable[[Tensor], Tensor]
+DefaultGaussianSliceProfile = SliceGaussian(2.0)
+
+
+class SliceProjectionOp(LinearOperator):
+    matrix: torch.Tensor | None
+    matrix_adjoint: torch.Tensor | None
+
+    def __init__(
+        self,
+        input_shape: SpatialDimension[int],
+        slice_rotation: Rotation | None = None,
+        slice_shift: float | Tensor = 0.0,
+        slice_profile: TensorFunction | np.ndarray | NestedSequence[TensorFunction] | float = 1.0,
+        optimize_for: Literal['forward', 'adjoint', 'both'] = 'both',
+    ):
+        """Create a module that represents the 'projection' of a volume onto a plane.
+
+        This operation samples from a 3D Volume a slice with a given rotation and shift,
+        according to the slice_profile. It can, for example, be used to describe the slice
+        selection of a 2D MRI sequence from the 3D Volume.-
+
+        The projection will be done by sparse matrix multiplication.
+
+        Rotation, shift, and profile can have (multiple) batch dimensions. These dimensions will
+        be broadcasted to a common shape and added to the front of the volume.
+        Different settings for different volume batches are NOT supported, consider creating multiple
+        operators if required,
+
+
+        Parameters
+        ----------
+        input_shape:
+            Shape of the 3D volume to sample from
+        slice_rotation
+            Rotation that describes the orientation of the plane. If None,
+            an identity rotation is used.
+        slice_shift
+            Offset of the plane in the volume perpendicular plane from the center of the volume.
+        slice_profile:
+            A function that called with a distance x from the slice center should return the
```

Comment:
Maybe: A function that is called with a tensor describing the distance x relative to centre of the volume. It should return the slice profile (i.e. intensity) at the positions in x.